app/src/main/kotlin/org/equeim/tremotesf/ui/torrentslistfragment/TorrentsAdapter.kt

```diff
@@ -131,7 +132,16 @@ class TorrentsAdapter(private val fragment: TorrentsListFragment) :
                 }
                 etaTextView.text = FormatUtils.formatDuration(context, torrent.eta)
 
-                progressBar.progress = (torrent.percentDone * 100).toInt()
+                val progress = (torrent.percentDone * 100).toInt()
+                progressBar.progress = progress
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    progressBar.progressTintList = if (progress == 100) {
```

Comment:
AFAIK minimum API here is L, not M.
For pre-Lollipop versions we could probably use the same solution as in ProgressBar.fixPreLollipopColor() extension function in ui/utils/Extensions.kt (and make it more generic function).

Reply:
On the other hand, I think I'm ready to drop Android 4.x support.

Reply:
> AFAIK minimum API here is L, not M.

This is what Android Studio told me... :man_shrugging: :sweat_smile: Happy to change it if you'd like.

> On the other hand, I think I'm ready to drop Android 4.x support.

So should I leave it as is?

Reply:
Also, we need to call `setProgressTintList()` only when `isFinished` changes, since AFAIK it can perform some expensive operations even when passed ColorStateList is the same as before. It is especially important since `update()` is called from `onBindViewHolder()` which will be called frequently when scrolling (we can just cache isFinished value inside TorrentsViewHolder).